source/common/router/config_impl.h

```diff
@@ -215,6 +216,59 @@ class RouteEntryImplBase : public RouteEntry, public Matchable, public RedirectE
     uint64_t default_;
   };
 
+  /**
+   * Route entry implementation for weighted clusters.
+   * RouteEntryImplBase holds one or more weighted cluster objects,
+   * where each object has a back pointer to the parent
+   * RouteEntryImplBase object. Almost all functions in this class
+   * forward calls back to the parent, with the exception of
+   * clusterName and routeEntry.
+   */
+  struct WeightedClusterEntry : public RouteEntry, public Route {
+  public:
+    WeightedClusterEntry(const RouteEntryImplBase* parent, const std::string runtime_key,
+                         Runtime::Loader& loader, const std::string name, uint64_t weight)
+        : parent_(parent), runtime_key_(runtime_key), loader_(loader), cluster_name_(name),
+          cluster_weight_(weight) {}
+
+    uint64_t clusterWeight() const {
+      return loader_.snapshot().getInteger(runtime_key_, cluster_weight_);
+    }
+
+    static const uint64_t MAX_CLUSTER_WEIGHT;
+
+    // Router::RouteEntry
+    const std::string& clusterName() const override { return cluster_name_; }
+
+    void finalizeRequestHeaders(Http::HeaderMap& headers) const override {
+      return parent_->finalizeRequestHeaders(headers);
+    }
+
+    Upstream::ResourcePriority priority() const override { return parent_->priority(); }
+    const RateLimitPolicy& rateLimitPolicy() const override { return parent_->rateLimitPolicy(); }
+    const RetryPolicy& retryPolicy() const override { return parent_->retryPolicy(); }
+    const ShadowPolicy& shadowPolicy() const override { return parent_->shadowPolicy(); }
+    std::chrono::milliseconds timeout() const override { return parent_->timeout(); }
+
+    const VirtualCluster* virtualCluster(const Http::HeaderMap& headers) const override {
+      return parent_->virtualCluster(headers);
+    }
+
+    const VirtualHost& virtualHost() const override { return parent_->virtualHost(); }
+
+    // Router::Route
+    const RedirectEntry* redirectEntry() const override { return nullptr; }
+    const RouteEntry* routeEntry() const override { return this; }
+
+  private:
+    const RouteEntryImplBase* parent_;
+    const std::string runtime_key_;
+    Runtime::Loader& loader_;
+    const std::string cluster_name_;
+    uint64_t cluster_weight_;
+  };
+  typedef std::unique_ptr<WeightedClusterEntry> WeightedClusterEntryPtr;
```

Comment:
I don't have anything typed out (sorry), and clang-format doesn't check for this. You can look at the rest of the code base which in general is very consistent. I'm sorry that this is annoying but it's important to keep things consistent and you will pick it up pretty quickly.